app/stac_api/serializers.py

```diff
@@ -84,6 +86,36 @@ def update_or_create_links(model, instance, instance_type, links_data):
     )
 
 
+class CommonJsonValidatorSerializer(serializers.ModelSerializer):
+
+    def validate(self, attrs):
+
+        # see: https://www.django-rest-framework.org/api-guide/serializers/#object-level-validation
+
+        # common validation part for collection, item and asset:
+        expected_payload = list(self.fields.keys())
+        expected_payload_read_only = [
+            field for field in self.fields if self.fields[field].read_only
+        ]
+
+        for key in self.initial_data.keys():
+            if key not in expected_payload:
+                raise ValidationError(detail=_("Unexpected property in payload"), code=key)
+            if key in expected_payload_read_only:
+                raise ValidationError(detail=_("Found read-only property in payload"), code=key)
```

Comment:
This could have/should have been left in the validators as a function.

Reply:
moved it back ;-)